src/Process/TreeLayerProcessing.js

```diff
@@ -0,0 +1,10 @@
+export default function updateTreeLayer(processNodeFn, context, layer, node) {
+    if (!node) {
```

Comment:
If we don't force the timing by controlling when the method is called, I'll leave it completely to the user to make sure that its layer is fully ready.
I'll give it a try.